src/scripts/main.js

```diff
@@ -1,3 +1,285 @@
 'use strict';
 
-// write your code here
+let isStarted = false;
+let isWon = false;
+let isLose = false;
+const loseMessage = document.querySelector('.message-lose');
+const winMessage = document.querySelector('.message-win');
+const startMessage = document.querySelector('.message-start');
+const actionButton = document.querySelector('.start');
+const rows = [...document.querySelectorAll('.field-row')];
+const scoreTable = document.querySelector('.game-score');
+
+const cells = [...document.querySelectorAll('.field-cell')];
+const rowSize = 4;
+
+actionButton.addEventListener('click', () => {
+  isStarted = handleGameStatus(isStarted);
+});
+
+document.addEventListener('keydown', (e) => {
+  if (!isStarted) {
+    return;
+  }
+
+  if (e.key === 'ArrowRight') {
+    moveAllRight();
+  }
+
+  if (e.key === 'ArrowLeft') {
+    moveAllLeft();
+  }
+
+  if (e.key === 'ArrowDown') {
+    moveAllDown();
+  }
+
+  if (e.key === 'ArrowUp') {
+    moveAllUp();
+  }
+
+  checkForWin();
+  checkForLose();
+});
+
+const handleGameStatus = (isGameStarted) => {
+  if (isGameStarted || isWon || isLose) {
+    restartGame();
+  } else {
+    startGame();
+  }
+
+  return true;
+};
+
+const handleCellClasses = (element, value, action) => {
+  switch (action) {
+    case 'add':
```

Comment:
You reuse `'add'`,` 'delete'` and `'replace'` in some places. Think about move it in constants